dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePluginFunctionalTest.kt

```diff
@@ -0,0 +1,323 @@
+            .withProjectDir(testProjectDir)
+            .withArguments("--build-cache", "assemble")
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        assertThat(result.task(":assemble")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M fail W assembleDebug { configuration cache, checkProjectDependecies enabled }`() {
+        // The assembleRelease fails for configuration cache and not because of our
+        // code but because of these tasks: collect[Flavour1][Flavour2]ReleaseDependencies
+
+        // This test should fail because our `DdCheckSdkDepsTask` is not following the
+        // --configuration-cache best practices (we are using the project inside) and so far we
+        // could not easily solve that.
+
+        // Given
+        stubGradleBuildFromResourceFile("build_with_datadog_dep.gradle.kts")
+
+        // When
+        assertThatThrownBy {
+            GradleRunner.create()
+                .withProjectDir(testProjectDir)
+                .withArguments("--configuration-cache", "assemble")
+                .withPluginClasspath(getTestConfigurationClasspath())
+                .build()
+        }.isInstanceOf(UnexpectedBuildFailure::class.java)
+    }
+
+    @Test
+    fun `M success W assembleDebug { configuration cache, checkProjectDependencies disabled }`() {
+
+        // Given
+        stubGradleBuildFromResourceFile("build_with_check_deps_disabled.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("--configuration-cache", "assembleDebug")
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        assertThat(result.task(":assembleDebug")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M success W assemble { Datadog SDK not in deps, checkProjectDepedencies set to warn }`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_check_deps_set_to_warn.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("assemble")
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        assertThat(result.task(":assemble")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M fail W assemble { Datadog SDK not in deps }`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_without_datadog_dep.gradle.kts")
+        // When
+        assertThatThrownBy {
+            GradleRunner.create()
+                .withProjectDir(testProjectDir)
+                .withArguments("assemble")
+                .withPluginClasspath(getTestConfigurationClasspath())
+                .build()
+        }.isInstanceOf(UnexpectedBuildFailure::class.java)
```

Comment:
should we check that cause is `MissingSdkException` or it is not present in the exception thrown?

Reply:
Could not see it in the thrown exception but will have another look.

Reply:
As I explained also for Xavier's comment we are not able to get the root exception as GradleRunner is actually intercepting it under the hood but not passing it along with the result or in the thrown UnexpectedBuildFailure.